### PR TITLE
Update to enable the use of the preliminary modular environment

### PR DESCRIPTION
**Pull request by ens-sb**
## Description

**Related JIRA tickets:**
- ENSCOMPARASW-7584

#### Change 1
Updated SLURM json config to use software from the prod-base spack environment if available.

#### Change 2
Disabled cheking for BLAST database files which arecno longer present with the new version.

#### Change 3
Fixed the module path for Bio::Tools::Phylo::PAML::Codeml.

## Testing
The metazoa insects protein trees pipeline was run to completion.

---

For code reviewers: [code review SOP](https://www.ebi.ac.uk/seqdb/confluence/display/EnsCom/Code+review+SOP)

### CODE REVIEW

File: conf/software/SLURM.standard*.json [hunk 1]
```diff
@@ -4,76 +4,76 @@
...
+    "axtChain_exe"              : "/hps/software/users/ensembl/ensw/swenv/env/prod-base/bin/axtChain",
+    "bedtools_exe"              : ["check_exe_in_cellar","bedtools/2.30.0/bin/bedtools"],
+    "big_bed_exe"               : "/hps/software/users/ensembl/ensw/swenv/env/prod-base/bin/bedToBigBed",
+    "big_wig_exe"               : "/hps/software/users/ensembl/ensw/swenv/env/prod-base/bin/bedGraphToBigWig",
     "bl2seq_exe"                : null,
-    "blast_bin_dir"             : ["check_dir_in_cellar", "blast/2.2.30/bin"],
-    "blastn_exe"                : ["check_exe_in_cellar", "blast/2.2.30/bin/blastn"],
-    "blat_exe"                  : ["check_exe_in_cellar", "kent/v335_1/bin/blat"],
+    "blast_bin_dir"             : "/hps/software/users/ensembl/ensw/swenv/env/prod-base/bin",
+    "blastn_exe"                : "/hps/software/users/ensembl/ensw/swenv/env/prod-base/bin/balstn",
```
Comment (twalsh-ebi) [inline conf/software/SLURM.standard*.json:13]:
```suggestion
    "blastn_exe"                : "/hps/software/users/ensembl/ensw/swenv/env/prod-base/bin/blastn",
```

**Review by Simarpreet-Kaur-Bhurji — APPROVED** [review]

File: modules/Bio/EnsEMBL/Compara/RunnableDB/BlastAndParsePAF.pm [hunk 2]
```diff
@@ -122,7 +122,7 @@ sub fetch_input {
         #                           unannotated.fasta.01.pog
         my $fastafile = $self->param('blast_db');
 
-        foreach my $ext (qw(phr pin pog psd psi psq)) {
+        foreach my $ext (qw(phr pin pog psq)) {
```
Comment (twalsh-ebi) [inline modules/Bio/EnsEMBL/Compara/RunnableDB/BlastAndParsePAF.pm:125]: Does `makeblastdb` now generate only files with these extensions, or are there additional BLAST database files with suffixes not listed here?

File: conf/software/SLURM.standard*.json [hunk 3]
```diff
@@ -4,76 +4,76 @@
...
-    "raxml_exe_sse3"            : ["check_exe_in_cellar", "raxml/8.2.8/bin/raxmlHPC-SSE3"],
-    "raxml_pthread_exe_avx"     : ["check_exe_in_cellar", "raxml/8.2.8/bin/raxmlHPC-PTHREADS-AVX"],
-    "raxml_pthread_exe_sse3"    : ["check_exe_in_cellar", "raxml/8.2.8/bin/raxmlHPC-PTHREADS-SSE3"],
-    "samtools_exe"              : ["check_exe_in_cellar", "samtools/1.12/bin/samtools"],
-    "semphy_exe"                : ["check_exe_in_cellar", "semphy/2.0b3/bin/semphy"],
-    "server_exe"                : ["check_exe_in_cellar", "exonerate24/2.4.0/bin/exonerate-server"],
-    "treebest_exe"              : ["check_exe_in_compara", "treebest/rc5/bin/treebest"],
+    "fasta2esd_exe"             : "/hps/software/users/ensembl/ensw/swenv/env/prod-base/bin/fasta2esd",
+    "fasttree_exe"              : ["check_exe_in_cellar","fasttree/2.1.8/bin/FastTree"],
+    "faToNib_exe"               : "/hps/software/users/ensembl/ensw/swenv/env/prod-base/bin/faToNib_exe",
```
Comment (twalsh-ebi) [inline conf/software/SLURM.standard*.json:34]:
```suggestion
    "faToNib_exe"               : "/hps/software/users/ensembl/ensw/swenv/env/prod-base/bin/faToNib",
```

File: modules/Bio/EnsEMBL/Compara/RunnableDB/BlastAndParsePAF.pm (hunk 2 above)
Comment (ens-sb) [reply modules/Bio/EnsEMBL/Compara/RunnableDB/BlastAndParsePAF.pm:125]: Let us handle this issue in a different mini-ticket.

Comment (twalsh-ebi) [reply modules/Bio/EnsEMBL/Compara/RunnableDB/BlastAndParsePAF.pm:125]: OK: ENSCOMPARASW-7616

**Review by twalsh-ebi — APPROVED** [review]

**Pull request merged**